app/api/domains/api.py

```diff
@@ -108,6 +109,38 @@ def format_map_basic(m: Beatmap) -> dict[str, object]:
     }
 
 
+@router.get("/search_players")
+async def api_search_players(
+    search: Optional[str] = Query(None, alias="q", min=2, max=32),
+    db_conn: databases.core.Connection = Depends(acquire_db_conn),
+):
+    """Search for users on the server by name."""
+
+    # execute the query using the database connection
```

Comment:
i used to put comments like this all over my code, but my new mindset is https://www.youtube.com/watch?v=2a_ytyt9sf8&t=626s

Reply:
the code here is actually very self-expressive though, i think it's quite readable without the comments

Reply:
while i don't agree with this mindset, i will apply it here

Reply:
would recommend actually watching into the vid though, he explains it more in-depth - e.g. refactoring your code to read *as if it were a comment*;
![image](https://i.imgur.com/NdEPMP0.png)

Reply:
this makes more sense, i'll watch the video when i get a chance